modules/azure/loganalytics.go

```diff
@@ -0,0 +1,93 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2020-03-01-preview/operationalinsights"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// LogAnalyticsWorkspaceExists indicates whether the operatonal insights workspaces exists.
+// This function would fail the test if there is an error.
+func LogAnalyticsWorkspaceExists(t testing.TestingT, workspaceName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := LogAnalyticsWorkspaceExistsE(workspaceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// GetLogAnalyticsWorkspaceSku return the log analytics workspace SKU as string as one of the following: Free, Standard, Premium, PerGB2018, CapacityReservation; otherwise empty string "".
+// This function would fail the test if there is an error.
+func GetLogAnalyticsWorkspaceSku(t testing.TestingT, workspaceName string, resourceGroupName string, subscriptionID string) string {
+	sku, err := GetLogAnalyticsWorkspaceSkuE(workspaceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return string(sku)
+}
+
+func GetLogAnalyticsWorkspaceSkuE(workspaceName string, resourceGroupName string, subscriptionID string) (string, error) {
+	ws, err := GetLogAnalyticsWorkspaceE(workspaceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	return string(ws.Sku.Name), nil
+}
+
+// GetLogAnalyticsWorkspaceRetentionPeriodDays returns the log analytics workspace retention period in days.
+// This function would fail the test if there is an error.
+func GetLogAnalyticsWorkspaceRetentionPeriodDays(t testing.TestingT, workspaceName string, resourceGroupName string, subscriptionID string) int32 {
```

Comment:
Remove this function, we want to keep the helpers slim to those that really simplify retrieving data from azure.  Keep the int test but modify to get off client object.